src/plots/cartesian/axis_ids.js

```diff
@@ -18,7 +18,7 @@ var constants = require('./constants');
 // completely in favor of just 'x' if it weren't ingrained in the API etc.
 exports.id2name = function id2name(id) {
     if(typeof id !== 'string' || !id.match(constants.AX_ID_PATTERN)) return;
-    var axNum = id.substr(1);
+    var axNum = id.split(' ')[0].substr(1);
```

Comment:
Hmm... not sure about this one.
What happens if one used spaces in the axis name?
We should still be able to handle that case.

Reply:
you mean something like `'x 2'`? I think that would get caught by the reference coercion, no? IIRC @alexcjohnson  was saying that here we could assume `id` was valid and had already been "cleaned".